src/handlers/decision.rs

```diff
@@ -152,6 +164,164 @@ pub(super) async fn handle_command(
+        let expected_comment = "| Team member | State |\n\
+        |-------------|-------|\n\
+        | Barbara | ~~hold~~  ~~merge~~  **merge** |\n\
+        | Niklaus | ~~merge~~  ~~hold~~  **merge** |"
+            .to_string();
+
+        assert_eq!(build_result, expected_comment);
+    }
+
+    #[test]
+    fn test_successfuly_build_comment_user_no_votes() {
+        let mut history: BTreeMap<String, Vec<UserStatus>> = BTreeMap::new();
+        let mut current_statuses: BTreeMap<String, Option<UserStatus>> = BTreeMap::new();
+
+        // user 1
+        let mut user_1_statuses: Vec<UserStatus> = Vec::new();
+        user_1_statuses.push(create!(UserStatus));
+        user_1_statuses.push(create!(UserStatus, :hold));
+
+        history.insert("Niklaus".to_string(), user_1_statuses);
+
+        current_statuses.insert("Niklaus".to_string(), Some(create!(UserStatus)));
+
+        // user 2
+        let mut user_2_statuses: Vec<UserStatus> = Vec::new();
+        user_2_statuses.push(create!(UserStatus, :hold));
+        user_2_statuses.push(create!(UserStatus));
+
+        history.insert("Barbara".to_string(), user_2_statuses);
+
+        current_statuses.insert("Barbara".to_string(), Some(create!(UserStatus)));
+
+        // user 3
+        history.insert("Tom".to_string(), Vec::new());
+
+        current_statuses.insert("Tom".to_string(), None);
+
+        let build_result = build_status_comment(&history, &current_statuses)
+            .expect("it shouldn't fail building the message");
+        let expected_comment = "| Team member | State |\n\
+        |-------------|-------|\n\
+        | Barbara | ~~hold~~  ~~merge~~  **merge** |\n\
+        | Niklaus | ~~merge~~  ~~hold~~  **merge** |\n\
+        | Tom |  |"
+            .to_string();
+
+        assert_eq!(build_result, expected_comment);
+    }
+
+    #[test]
+    fn test_build_comment_inconsistent_users() {
+        let mut history: BTreeMap<String, Vec<UserStatus>> = BTreeMap::new();
+        let mut current_statuses: BTreeMap<String, Option<UserStatus>> = BTreeMap::new();
+
+        // user 1
+        let mut user_1_statuses: Vec<UserStatus> = Vec::new();
+        user_1_statuses.push(create!(UserStatus));
+        user_1_statuses.push(create!(UserStatus, :hold));
+
+        history.insert("Niklaus".to_string(), user_1_statuses);
+
+        current_statuses.insert("Niklaus".to_string(), Some(create!(UserStatus)));
+
+        // user 2
+        let mut user_2_statuses: Vec<UserStatus> = Vec::new();
+        user_2_statuses.push(create!(UserStatus, :hold));
+        user_2_statuses.push(create!(UserStatus));
+
+        history.insert("Barbara".to_string(), user_2_statuses);
+
+        current_statuses.insert("Martin".to_string(), Some(create!(UserStatus)));
+
+        let build_result = build_status_comment(&history, &current_statuses);
+        assert_eq!(
+            format!("{}", build_result.unwrap_err()),
+            "user Barbara not present in current statuses list"
```

Comment:
shorthand for testing the root cause of the error, without doing downcast_ref and such.